distributed/deploy/tests/test_local.py

```diff
@@ -46,11 +47,36 @@ def test_procs(loop):
         with Client(c.scheduler.address, loop=loop) as e:
             assert all(v == 3 for v in e.ncores().values())
 
-            c.start_worker(nanny=False)
-            assert isinstance(c.workers[-1], Worker)
+            c.start_worker()
+            assert all(isinstance(w, Nanny) for w in c.workers)
         repr(c)
 
 
+def test_transports():
+    """
+    Test the transport chosen by LocalCluster depending on arguments.
+    """
+    with LocalCluster(1, nanny=False, silence_logs=False) as c:
+        assert c.scheduler_address.startswith('inproc://')
+        assert c.workers[0].address.startswith('inproc://')
+        with Client(c.scheduler.address) as e:
+            assert e.submit(inc, 4).result() == 5
```

Comment:
Sorry for the delay in reviewing.  I thought I had pressed the "submit review" button hours ago...